### PR TITLE
Add all missing fields to ResponseItem and related classes.

### LINKED ISSUE #322: jackson marshalling exception with ResponseItem.ProgressDetail
I'm getting the following exception with version 2.1.0 of docker-java

com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "start" (class com.github.dockerjava.api.model.ResponseItem$ProgressDetail), not marked as ignorable (2 known properties: , "total", "current"])

Looking at the code, it appears the fields are annotated with 'ignoreUnknown = false', yet there seems to be an unknown 'start' field coming back from the docker registry.  Our registry is artifactory, btw, not the docker-registry or docker.io, so it may be sending back some extra fields to be 'helpful'.

This annotation seems too strict?

Interesting, this error does *not* occur when running on my OS X laptop.  It only happens on the build machine running an old version of jenkins, java and linux.  Unfortunately, getting them to upgrade can take months.

Any ideas?

### PR DESCRIPTION
**Pull request by llamahunter**
Fix types to match Go code.

fixes #322

### CODE REVIEW

File: src/main/java/com/github/dockerjava/api/model/PushResponseItem.java [hunk 1]
```diff
@@ -18,7 +18,7 @@
      */
     @JsonIgnore
     public boolean isErrorIndicated() {
-        if (getError() == null)
+        if (getError() == null && getErrorDetail() == null)
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/api/model/PushResponseItem.java]: are you sure that error can't exist without error details? Shouldn't be it ORed?

File: src/main/java/com/github/dockerjava/api/model/ResponseItem.java [hunk 2]
```diff
@@ -16,57 +16,83 @@
...
+    public String getStatus() {
+        return status;
     }
 
     public ProgressDetail getProgressDetail() {
         return progressDetail;
     }
 
+    @Deprecated
+    public String getProgress() {
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/api/model/ResponseItem.java]: Deprecate also corresponding field?

Comment (llamahunter) [inline src/main/java/com/github/dockerjava/api/model/ResponseItem.java]: It's private, so don't really care as much.

File: src/main/java/com/github/dockerjava/api/model/PushResponseItem.java (hunk 1 above)
Comment (llamahunter) [inline src/main/java/com/github/dockerjava/api/model/PushResponseItem.java]: If either is not null, its an error.  Only when both are null (i.e. 'and') is it safe to return false.

Comment (marcuslinke) [inline src/main/java/com/github/dockerjava/api/model/PushResponseItem.java]: I think this method should be pushed up to `ResponseItem` class. Then it should be used in `BuildResponseItem.isBuildSuccessIndicated()` and `PullResponseItem.isPullSuccessIndicated()` as pre-condition check.

Comment (llamahunter) [inline src/main/java/com/github/dockerjava/api/model/PushResponseItem.java]: Such refactoring is exceeding the scope of this pull request, which is to get docker-java to stop rejecting valid ProgressDetail messages.

**Comment by marcuslinke** [comment]
@llamahunter Thanks for your effort!

File: src/main/java/com/github/dockerjava/api/model/ResponseItem.java (hunk 2 above)
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/api/model/ResponseItem.java]: Not javadoc why it deprecated and what to use

Comment (llamahunter) [inline src/main/java/com/github/dockerjava/api/model/ResponseItem.java]: Talk to the docker people. Anyway, it's kind of obvious what the replacement is, no? ProgressDetail.

Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/api/model/ResponseItem.java]: No, not obvious when you don't remember all details of constantly changing API. Back to this place third time.

**Comment by KostyaSha** [comment]
> Ok... refactored the isErrorIndicated() method and updated build and pull response items.

Now it uses deprecated method that looks weird.

**Comment by KostyaSha** [comment]
Where did you get doc that describes this responses? :(

**Comment by marcuslinke** [comment]
@KostyaSha This is documented in the related issue #322

**Pull request merged**